### PR TITLE
Spam safelisting improvements

### PR DESCRIPTION
**Pull request by mitsosf**

### CODE REVIEW

File: zenodo/modules/spam/tasks.py [hunk 1]
```diff
@@ -74,8 +79,9 @@ def delete_spam_user(user_id, deleted_by):
             c.delete()
     current_accounts.datastore.deactivate_user(user)
     db.session.commit()
-    for r in rs.scan():
-        delete_record(r.meta.id, 'spam', deleted_by)
+    record_ids = [record.meta.id for record in rs.scan()]
+    for record_id in record_ids:
+        async_delete_record(record_id, 'spam', deleted_by)
```
Comment (slint) [inline zenodo/modules/spam/tasks.py:84]:
```suggestion
        async_delete_record.delay(record_id, 'spam', deleted_by)
```

File: zenodo/modules/spam/tasks.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def check_metadata_for_spam(community_id=None, dep_id=None):
     return spam_proba
 
 
+@shared_task(ignore_result=False)
+def async_delete_record(record_uuid, reason, user):
+    delete_record(record_uuid, reason, user)
```
Comment (slint) [inline zenodo/modules/spam/tasks.py:61]:
```suggestion
    from zenodo.modules.deposit import utils as deposit_utils
    ...
    deposit_utils.delete_record(record_uuid, reason, user)
```

File: zenodo/modules/spam/utils.py [hunk 3]
```diff
@@ -75,20 +82,22 @@ def send_spam_admin_email(user, deposit=None, community=None):
...
-                task = check_metadata_for_spam.delay(dep_id=str(deposit.id))
-                user_id = deposit['owners'][0]
-            spam_proba = task.get(timeout=current_app.config[
-                'ZENODO_SPAM_CHECK_TIMEOUT'])
-        else:
-            spam_proba = 0
-        if spam_proba > current_app.config['ZENODO_SPAM_THRESHOLD']:
-            if not Permission(ActionNeed('admin-access')).can():
+        if not is_safelisted_user(current_user) or \
+            not Permission(ActionNeed('admin-access')).can():
```
Comment (slint) [inline zenodo/modules/spam/utils.py:86]:
```suggestion
        is_admin = Permission(ActionNeed('admin-access')).can()
        if not is_safelisted_user(current_user) or \
            not is_admin:
```

And further for readability

File: zenodo/modules/spam/templates/zenodo_spam/safelist/admin.html [hunk 4]
```diff
@@ -52,6 +52,14 @@ <h1>Safelist admin</h1>
                  ('max_users') }}">
         </div>
       </div>
+      <select name="include_pending" class="form-control">
+        <option id="1" {{ "selected" if  request.args.get
```
Comment (slint) [inline zenodo/modules/spam/templates/zenodo_spam/safelist/admin.html:56]:
```suggestion
        <option value="1" {{ "selected" if  request.args.get
```

maybe works with GET

File: zenodo/modules/spam/views.py [hunk 5]
```diff
@@ -194,9 +199,10 @@ def safelist_admin():
             'last_records': ", ".join(r.title for r in user.records),
             'last_descriptions': ", ".join(r.description for r in
                                            user.records),
-            'first_record_id': user.records[0].recid
+            'first_record_id': user.records[0].recid,
+            'total_records': user.doc_count()
```
Comment (slint) [inline zenodo/modules/spam/views.py:203]:
```suggestion
            'total_records': user.doc_count
```

File: zenodo/modules/spam/views.py [hunk 6]
```diff
@@ -152,7 +153,8 @@ def _expand_users_info(results):
             "id": user.id,
             "email": user.email,
             "external": [i.method for i in (user.external_identifiers or [])],
-            "flag": _evaluate_user_domain(user.email)
+            "flag": _evaluate_user_domain(user.email),
+            "user_status": user.active
```
Comment (slint) [inline zenodo/modules/spam/views.py:157]:
```suggestion
            "active": user.active,
            "safelist": user.safelist,
```

File: zenodo/modules/spam/templates/zenodo_spam/safelist/admin.html (hunk 4 above)
Comment (mitsosf) [reply zenodo/modules/spam/templates/zenodo_spam/safelist/admin.html:56]: It does work with GET btw :)

**Review by slint — APPROVED** [review]

**Pull request merged**